docs/design-go-api.md

```diff
@@ -0,0 +1,218 @@
+    WithUseTLS(true)
+
+var client *glide.RedisClusterClient
+var err error
+client, err = glide.CreateClusterClient(config)
+```
+
+### Case 5: Get(key) from connected RedisClient
+```go
+result, err := client.Get("apples")
+```
+
+### Case 6: Set(key, value) from connected RedisClient
+```go
+// Without setOptions
+err := client.Set("apples", "oranges")
+
+// With setOptions
+var setOptions *glide.SetOptions
+setOptions = glide.NewSetOptions().
+    WithReturnOldValue(true)
+oldValue, err := client.SetWithOptions("apples", "oranges", setOptions)
+```
+
+### Case 7: Get(key) from a disconnected RedisClient
+Return a glide.ConnectionError if the RedisClient is closed/disconnected
+```go
+result, err := client.Get("apples")
+if err != nil {
+    connErr, isConnErr := err.(glide.ConnectionError)
+    if isConnErr {  
+        log.Fatal("RedisClient get failed with: " + connErr.Error())
+    }
+}
+```
+
+### Case 8: Send custom command to RedisClient
+```go
+var result interface{}
+var err error
+
+result, err = client.CustomCommand([]{"GET", "apples"})
+if err != nil {
+    log.Fatal("RedisClient failed to execute custom command with: " + err.Error())
+}
+
+strResult, isString := result.(string)
+if !isString {
+    log.Fatal("Expected result to be of type string but the actual type was: " + reflect.TypeOf(result))
+}
+```
+
+### Case 9: Send transaction to RedisClient
+```go
+transaction := glide.NewTransaction()
+transaction.Get("apples")
+transaction.Get("pears")
+transaction.Set("cherries", "Bing")
+
+var result []interface{}
+var err error
+result, err = client.Exec(transaction)
+if err != nil {
+    log.Fatal("Redis client transaction failed with: " + err.Error())
+}
+
+firstResponse := result[0]
+secondResponse := result[1]
+thirdResponse := result[2]  // evaluates to nil
+```
+
+### Case 10: Send Get request to a RedisClusterClient with one address
+```go
+var config *glide.ClusterClientConfiguration
+config = glide.NewClusterClientConfiguration().
+    WithAddress(glide.AddressInfo{Host: host, Port: port}).
+    WithUseTLS(true)
+
+// Create a client and connect
+var client *glide.RedisClusterClient
+var err error
+client, err = glide.CreateClusterClient(config)
+if err != nil {
+    log.Fatal("Redis client failed with: " + err.Error())
+}
+
+result, err := client.Get("apples")
+```
+
+### Case 11: Send Get request to a RedisClusterClient with multiple addresses
+```go
+var config *glide.ClusterClientConfiguration
+config = glide.NewClusterClientConfiguration().
+    WithAddress(glide.AddressInfo{Host: host1, Port: port1}).
+    WithAddress(glide.AddressInfo{Host: host2, Port: port2}).
+    WithUseTLS(true)
+
+// Create a client and connect
+var client *glide.RedisClusterClient
+var err error
+client, err = glide.CreateClusterClient(config)
+if err != nil {
+    log.Fatal("Redis client failed with: " + err.Error())
+}
+
+result, err := client.GetWithRoutes("apples", glide.NewRoute(glide.AllNodes))
```

Comment:
I'm not sure what the best way is to represent routes in Go, I'm wondering if using a builder like java would be best. This is how I represented it for now